functions/src/test/notifications.test.ts

```diff
@@ -0,0 +1,173 @@
+import * as assert from 'assert'
+import * as sinon from 'sinon'
+import * as mocks from './mocks'
+import 'mocha'
+
+import * as notifications from '../notifications'
+
+// NOTE: The `@ts-ignore` statements in this file are because much of the data
+// does not fit the shape of the expected data. This is what we want, because
+// we're trying to test the edge cases where the data doesn't fit what's expected.
+
+describe('Activity creation handler', () => {
+    it('Gets activity data based on the data changed', async () => {
+        const dataSpy = sinon.spy(notifications, 'getActivityData')
+        // @ts-ignore
+        await notifications.activityCreationHandler(mocks.testActivitySnap, {})
+        assert(dataSpy.calledOnce)
```

Comment:
Here we spy on the export of the other file to make sure it was called by the function